framework/include/base/Assembly.h

```diff
@@ -214,7 +214,7 @@ class Assembly
    * Returns the reference to the current neighbor volume
    * @return A _reference_.  Make sure to store this as a reference!
    */
-  const Real & neighborVolume() { return _current_neighbor_volume; }
+  const Real & neighborVolume() { _need_neighbor_elem_volume = true; return _current_neighbor_volume; }
```

Comment:
I suppose this is ok. This method is probably getting big enough to move to the .C file.